src/core/Form/SearchInput/SearchInput.tsx

```diff
@@ -3,47 +3,75 @@ import { default as styled } from 'styled-components';
 import { InternalTokensProp } from '../../theme';
 import { withSuomifiDefaultProps } from '../../theme/utils';
 import { baseStyles } from './SearchInput.baseStyles';
-import { TextInput, TextInputProps } from '../TextInput/TextInput';
+import { baseStyles as inputBaseStyles } from '../TextInput/TextInput.baseStyles';
+import { BaseTextInput } from '../../../components/Form/TextInput';
+import { TextInputProps, textInputClassNames } from '../TextInput/TextInput';
 import { Icon } from '../../Icon/Icon';
 import classnames from 'classnames';
 import { Omit } from '../../../utils/typescript';
+import { logger } from '../../../utils/logger';
 
 const baseClassName = 'fi-search-input';
 const inputContainerBaseClassName = `${baseClassName}_input-container`;
 const inputBaseClassName = `${baseClassName}_input`;
 const iconBaseClassName = `${baseClassName}_icon`;
 
-export interface SearchInputProps extends Omit<TextInputProps, 'variant'> {}
+export interface SearchInputProps extends Omit<TextInputProps, 'variant'> {
+  /** Placeholder text for search field. Must match labelText */
+  placeholder?: string;
+}
 
 const StyledTextInput = styled(
   ({
     tokens,
     className,
+    labelTextProps = { className: undefined },
+    inputContainerProps = { className: inputContainerBaseClassName },
     inputClassName,
     ...passProps
   }: TextInputProps & InternalTokensProp) => (
-    <TextInput
+    <BaseTextInput
+      {...passProps}
+      labelTextProps={{
+        ...labelTextProps,
+        className: classnames(
+          labelTextProps.className,
+          textInputClassNames.labelParagraph,
+        ),
+      }}
+      inputContainerProps={{
+        ...inputContainerProps,
+        className: classnames(
+          inputContainerProps.className,
+          textInputClassNames.inputContainer,
+        ),
+      }}
       {...passProps}
       className={classnames(className, baseClassName)}
       inputClassName={classnames(inputClassName, inputBaseClassName)}
-      inputContainerProps={{ className: inputContainerBaseClassName }}
     />
   ),
 )`
   ${props => baseStyles(props)}
+  ${props => inputBaseStyles(props)}
```

Comment:
I think this should go first above the baseStyles given here (CSS definitions order overriding).